src/pyHexDump/cmd_print_checksum.py

```diff
@@ -133,57 +66,74 @@ def cmd_print_checksum(binary_file, start_address, end_address, seed):
         end_address (int):  Address where to end the calculation (not included)
-        seed (int): Seed value for the CRC calculation
+        seed (int): Seed value for the CRC calculation (default 0)
+        polynomial(int): Generator polynomial to use in the CRC calculation.
+                         The bits in this integer are to coefficients in the polynomial.
+        bit_width(int): Number of bits for the CRC calculation. They have to match with
+                        the generator polynomial
+        reverse_input(bool): Reflect each single input byte if True
+        reverse_output(bool): Reflect the final CRC value if True
+        final_xor(bool): Xor the final result with the value 0xff before returning the soulution
 
     Returns:
         Ret, checksum: Statusinformation and checksum
     """
     ret_status = Ret.OK
-    mem_access = MemAccessU32LE()
+    mem_access = MemAccessU8()
 
     offset = 0
-    count = (end_address-start_address)/4
-    # The number to calculate the CRC has to be a multiple of 4
-    if (end_address-start_address)%4 != 0:
+    count = (end_address - start_address) / mem_access.get_size()
+
+    # The number to calculate the CRC has to be a multiple of word size
+    if (end_address - start_address) % mem_access.get_size() != 0:
         ret_status = Ret.ERROR_CRC_CACLULATION
         return ret_status, None
 
-    # Use ctypes that Python behaves like C
-    crc = ctypes.c_uint32(seed)
-    # Word which is currently calculated
-    word = ctypes.c_uint32(0)
-    # Value for the crc tabular
-    tab_index = ctypes.c_uint32(0)
-    # Value to calculate the left shift
-    left_shift = ctypes.c_uint32(0)
+    bit_width_mask = pow(2, bit_width) - 1
+    msb_mask = 1 << bit_width
+    crc = seed
+    word = 0
+
+    polynomial = (1 << bit_width) | polynomial
 
     while count > 0:
         count -= 1
 
-        word.value = mem_access.get_value(binary_data, start_address+offset)
+        word = mem_access.get_value(binary_data, start_address + offset)
+        if reverse_input:
+            tmp = '{:0{width}b}'.format(word, width=8)
+            word = int(tmp[::-1], 2)
+
+
+        crc = crc ^ (word << (bit_width - 8))
+
+        for _ in range(8):
+            crc = crc << 1
 
-        # Do the following calculation step by step to ensure that Python behaves like C
-        #crc = crc32_tab[((crc>>24) ^ word) & 0xFF] ^ (crc<<8)
-        tab_index.value = crc.value>>24
-        tab_index.value = tab_index.value ^ word.value
-        tab_index.value = tab_index.value & 0xFF
+            if (crc & msb_mask) != 0:
+                crc = crc ^ polynomial
 
-        left_shift.value = crc.value<<8
+        offset += mem_access.get_size()
 
-        crc.value = crc32_tab[tab_index.value] ^ left_shift.value
+    crc &= bit_width_mask
 
-        offset += 1
+    if reverse_output:
+        tmp = '{:0{width}b}'.format(word, width=bit_width)
+        word = int(tmp[::-1], 2)
```

Comment:
The reverse_output feature uses "word" instead of "crc". I fixed that in the main branch.